moduleframework/mtf_scheduler.py

```diff
@@ -0,0 +1,176 @@
+        os.environ['CONFIG']=args.config
+    if args.url:
+        os.environ['URL']=args.url
+    if args.modulemdurl:
+        os.environ['MODULEMDURL']=args.modulemdurl
+
+    return args, unknown
+
+class AvocadoStart(object):
+    def __init__(self, args, unknown):
+
+        self.module = os.environ['MODULE']
+        self.action = args.action
+
+        # its used for filepath in loadCli:
+        self.tmppath = tempfile.mktemp()
+        site_libs = os.path.dirname(moduleframework.__file__)
+        mtf_tools = "tools"
+
+        # choose between TESTS and ADDITIONAL ENVIRONMENT from options
+        self.tests = ''
+        if args.linter:
+            self.tests = ("{SITE_LIB}/{MTF_TOOLS}/*.py".format(SITE_LIB=site_libs, MTF_TOOLS=mtf_tools))
+        self.additionalAvocadoArg = ''
+        self.returncode=0
+
+        for param in unknown:
+            # take care of this, see tags for safe/unsafe:
+            # http://avocado-framework.readthedocs.io/en/52.0/WritingTests.html#categorizing-tests
+            if os.path.exists(param):
+                # this is list of tests in local file
+                self.tests += " {0} ".format(param)
+            else:
+                # this is additional avocado param
+                self.additionalAvocadoArg += " {0} ".format(param)
+
+        common.print_debug("tests = {0}".format(self.tests))
+        common.print_debug("additionalAvocadoArg = {0}".format(self.additionalAvocadoArg))
+
+    def bash_command(self, cmd):
+        # only to run bash command
+        common.print_debug("cmd: {0}".format(cmd))
+        return process.run(cmd, shell=True, ignore_status=True)
+
+    def avocado_run(self):
+        avocado_args="--json {JSON_LOG}".format(JSON_LOG=self.tmppath)
+        avocadoAction = "MODULE={MODULE} avocado {ACTION} {AVOCADO_ARGS}".format(MODULE=self.module, ACTION=self.action, AVOCADO_ARGS=avocado_args)
+
+        # run avocado with right cmd arguments
+        # bash = self.bash_command(("{AVOCADO} {a} {b}".format(AVOCADO=avocadoAction, a=self.additionalAvocadoArg, b=self.tests)))
+        bash = self.bash_command(("{AVOCADO} {a} {b}".format(AVOCADO=avocadoAction, a=self.additionalAvocadoArg, b=self.tests)))
+        self.returncode += bash.exit_status
+        common.print_info(bash.stdout, bash.stderr)
+
+    def show_error(self):
+        # show errors details and delete log file
+        if os.path.exists(self.tmppath):
+            bash = self.bash_command("mtf-log-parser {JSON_LOG}".format(JSON_LOG=self.tmppath))
+            common.print_info(bash.stdout, bash.stderr)
+            self.returncode += bash.exit_status
+            os.remove(self.tmppath)
+
+
+def main():
+    common.print_debug('verbose/debug mode')
+    args, unknown = cli()
+
+    if args.setup:
+        # mtfenvset need bash environment!
+        from moduleframework.mtf_environment import mtfenvset
+        mtfenvset()
+
+    a = AvocadoStart(args, unknown)
+    if args.action == 'run':
```

Comment:
so any other avocado action is not calle, although in avocado_run it use param `self.action`

Reply:
done